### PR TITLE
Add `CompilerGeneratedAttribute` to `EqualityContract` property

### PR DESCRIPTION
**Pull request by Youssef1313**
Closes https://github.com/dotnet/roslyn/issues/61991

### CODE REVIEW

**Comment by AlekseyTs** [comment]
I think we need a clear explanation why do we want to make this change. It is quite possible I missed something, but I did not find it in the referenced issue.

**Comment by AlekseyTs** [comment]
Done with review pass (commit 2)

File: src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordEqualityContractProperty.cs [hunk 1]
```diff
@@ -142,15 +152,6 @@ internal sealed class GetAccessorSymbol : SourcePropertyAccessorSymbol
             {
             }
 
-            internal override void AddSynthesizedAttributes(PEModuleBuilder moduleBuilder, ref ArrayBuilder<SynthesizedAttributeData> attributes)
```
[the remark this answers is not in the document]
Comment (Youssef1313) [reply src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordEqualityContractProperty.cs:145]: There shouldn't be any harm, it's just redundant. If the property is compiler-generated, then for sure the accessors are.

From the old PR:

image

---

This is also consistent with what's done in other compiler-generated symbols. For example:

https://github.com/dotnet/roslyn/blob/29a29d54e5ddd609650d47cb58318f874f0b99f0/src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedFieldSymbolBase.cs#L54

Here, a synthesized field will not have the attribute if the containing type already has it.

So the attribute is usually put on the outer most compiler-generated symbol.

**Comment by Youssef1313** [comment]
> I am not sure why this PR is supposed to close #61991. It looks like that issue discusses completely different scenario and, perhaps, should be closed "By Design" on its own. It is quite possible that the decision somehow contributes to a decision to make the change presented here, but they are still not the same.

Yeah the original issue can just be closed as by-design. It's just the conversation there led to revising where `CompilerGeneratedAttribute` is added.

File: src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordEqualityContractProperty.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (AlekseyTs) [reply src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordEqualityContractProperty.cs:145]: > If the property is compiler-generated, then for sure the accessors are.

I don't think this is a safe assumption in general. For C# compiler, sure. But in metadata accessors and properties are distinct and each has its own set of attributes. I can easily imagine a synthesized property that uses user-defined methods as accessors. 

> do not emit CompilerGenerated attributes for fields inside compiler generated types

This isn't the same though, I think. In the type system (and I am not talking about C# exclusively), a field is contained by a type, it cannot be defined if there is no type. So, it is relatively safe to assume that a user cannot define a field in a compiler generated type. Accessors, however, are not contained by a property (despite what C# syntax might lead you to believe), properties and accessors are siblings, they are just linked in a special way.

[the remark this answers is not in the document]
Comment (Youssef1313) [reply src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordEqualityContractProperty.cs:145]: Yeah I see that from the metadata perspective. I'll add the attribute on both the property and the accessors.

**Comment by AlekseyTs** [comment]
Done with review pass (commit 6). It looks like there are still tests that require a baseline adjustment.

**Review by AlekseyTs — APPROVED** [review]
LGTM (commit 7)

File: src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordEqualityContractProperty.cs [hunk 2]
```diff
@@ -85,6 +85,16 @@ protected override void ValidatePropertyType(BindingDiagnosticBag diagnostics)
             VerifyOverridesEqualityContractFromBase(this, diagnostics);
         }
 
+
```
Comment (jcouv) [inline src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordEqualityContractProperty.cs:88]: nit: extra blank line
```suggestion
```

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 8)

**Pull request merged**